lite/examples/video_classification/android/app/src/main/java/org/tensorflow/lite/examples/videoclassification/MainActivity.kt

```diff
@@ -0,0 +1,378 @@
+                val targetFpsMultiplier = MAX_CAPTURE_FPS.div(MODEL_FPS)
+                val targetCaptureFps = MODEL_FPS * targetFpsMultiplier
+                val builder = ImageAnalysis.Builder()
+                    .setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
+                val extender: Camera2Interop.Extender<*> = Camera2Interop.Extender(builder)
+                extender.setCaptureRequestOption(
+                    CaptureRequest.CONTROL_AE_TARGET_FPS_RANGE,
+                    Range(targetCaptureFps, targetCaptureFps)
+                )
+                val imageAnalysis = builder.build()
+
+                imageAnalysis.setAnalyzer(executor) { imageProxy ->
+                    processImage(imageProxy)
+                }
+
+                // Combine the ImageAnalysis and Preview into a use case group.
+                val useCaseGroup = UseCaseGroup.Builder()
+                    .addUseCase(preview)
+                    .addUseCase(imageAnalysis)
+                    .setViewPort(binding.preview.viewPort!!)
+                    .build()
+
+                // Bind use cases to camera.
+                cameraProvider.bindToLifecycle(
+                    this, cameraSelector, useCaseGroup
+                )
+
+            } catch (e: Exception) {
+                Log.e(TAG, "Use case binding failed.", e)
+            }
+
+        }, ContextCompat.getMainExecutor(this))
+    }
+
+    /**
+     * Run a frames received from the camera through the TFLite video classification pipeline.
+     */
+    private fun processImage(imageProxy: ImageProxy) {
+        // Ensure that only one frame is processed at any given moment.
+        synchronized(lock) {
+            val currentTime = SystemClock.uptimeMillis()
+            val diff = currentTime - lastInferenceStartTime
+
+            // Check to ensure that we only run inference at a frequency required by the
+            // model, within an acceptable error range (e.g. 10%). Discard the frames
+            // that comes too early.
+            if (diff * MODEL_FPS >= 1000 * (1 - MODEL_FPS_ERROR_RANGE)) {
+                lastInferenceStartTime = currentTime
+
+                val image = imageProxy.image
+                image?.let {
+                    videoClassifier?.let { classifier ->
+                        val imageBitmap = Bitmap.createBitmap(
+                            it.width,
+                            it.height,
+                            Bitmap.Config.ARGB_8888
+                        )
+                        CalculateUtils.yuvToRgb(image, imageBitmap)
+                        // Create a rotated version for portrait display.
+                        val rotateMatrix = Matrix()
+                        rotateMatrix.postRotate(
+                            imageProxy.imageInfo.rotationDegrees.toFloat()
+                        )
+                        val startCrop = (it.width - it.height) / 2
```

Comment:
If camera is rotated, will it be possible that width is smaller than height?

Reply:
Actually we don't need to crop the image as it'll be square cropped in the VideoClassifier. Therefore I removed the cropping logic here.